packages/core/src/base/staticClient.ts

```diff
@@ -19,7 +19,16 @@ export class ResponseError extends Error {
   }
 }
 
-async function getObjectFromResponse(response: Response): Promise<object> {
+/**
+ * Extracts the dto object from the given response object
+ * 
+ * @param response the response object containing the dto
+ * 
+ * @returns the dto wrapped in a promise
+ */
+export async function getObjectFromResponse(
```

Comment:
@monkey-jeff I made sure we do the same thing that the packages (minipass-fetch and node-fetch) do:
https://github.com/npm/minipass-fetch/blob/fa47f5f9e9059bb0872dba2b0abf49cde960e9d3/lib/body.js#L72
so there's no performance impact.
The problem with calling json() is that if it throws we can't tell if the body was empty or just bad JSON. We can't check after the fact because the exception doesn't include the text and the stream has already been consumed.
@blittle I'm pretty sure that my thought here was that if we have for example a DELETE that returns a 200 and an empty string instead of "{}", I didn't want the SDK to throw an error about malformed JSON which is what would happen.

Reply:
Thanks Brian! I'm happy with keeping it the way it is. @monkey-jeff if you are fine with keeping things the same, can you please approve the PR?